src/development/packages-and-plugins/developing-packages.md

```diff
@@ -296,6 +296,51 @@ Note that as shown here, an app-facing package can have
 some platforms implemented within the package,
 and others in endorsed federated implementations.
 
+#### Shared iOS and macOS implementations
+
+Many frameworks support both iOS and macOS with identical
+or mostly identical APIs, making it possible to implement
+some plugins for both iOS and macOS with the same codebase.
+Normally each platform's implementation is in its own
+folder, but the `sharedDarwinSource` option allows iOS
+and macOS to use the same folder instead:
+
+
+```yaml
+flutter:
+  plugin:
+    platforms:
+      ios:
+        pluginClass: HelloPlugin
+        sharedDarwinSource: true
+      macos:
+        pluginClass: HelloPlugin
+        sharedDarwinSource: true
+
+environment:
+  sdk: ">=2.19.0 <3.0.0"
+  # Flutter versions prior to 3.7 did not support the
+  # sharedDarwinSource option.
+  flutter: ">=3.7.0"
+```
+
+When `sharedDawninSource` is enabled, instead of
+an `ios` directory for iOS and a `macos` directory
+for macOS, both platforms use a shared `darwin`
+directory for all code and resources. When enabling
+this option, you will need to move any existing files
+from `ios` and `macos` to the shared directory. You
+will also need to update the podspec file to set the
```

Comment:
```suggestion also need to update the podspec file to set the
```

Reply:
What exactly do you want me to remove here? If I remove this line, it will read:

> You dependencies and deployment targets for both platforms, for example:

which isn't a sentence.

Reply:
Oh, I just realized you probably meant to remove "will" rather than the whole line. I'll do that; let me know if you had something else in mind.